### PR TITLE
And here's a kotlin converted version for consideration

### PR DESCRIPTION
**Pull request by makosblade**
Kotlinified version, basically no logic changes needed

### CODE REVIEW

**Review by katherinebernstein — APPROVED** [review]
I am happy about this but also mildly fearful

**Review by geekbeast — APPROVED** [review]

**Comment by geekbeast** [comment]
Same.

File: src/main/kotlin/DatastoreElasticsearchImpl.kt [hunk 1]
```diff
@@ -0,0 +1,1545 @@
...
+            securableObjectType: SecurableObjectType,
+            securableObjects: Iterable<*>
+    ): Boolean {
+        val indexName = indexNamesByObjectType[securableObjectType]
+        val typeName = typeNamesByIndexName[indexName]
+        return triggerIndex(indexName, typeName, securableObjects, getIdFnForType(securableObjectType))
+    }
+
+    //TODO: Seems dangerous and like we should delete?
+    fun clearAllData(): Boolean {
```
Comment (geekbeast) [inline src/main/kotlin/DatastoreElasticsearchImpl.kt:1350]: @katherinebernstein @makosblade Seems removable? I will merge if this build succeeds so just submit another PR when you see this tomorrow.

**Review by geekbeast — APPROVED** [review]

**Comment by makosblade** [comment]
🔥 🥳

File: src/main/kotlin/DatastoreElasticsearchImpl.kt (hunk 1 above)
Comment (katherinebernstein) [reply src/main/kotlin/DatastoreElasticsearchImpl.kt:1350]: I thought we had already removed this 🤔

Comment (katherinebernstein) [reply src/main/kotlin/DatastoreElasticsearchImpl.kt:1350]: Oh, it was removed from DatastoreElasticsearchImpl.java

**Pull request merged**